lib/logstash/inputs/tcp.rb

```diff
@@ -260,14 +260,24 @@ def handle_socket(socket)
   end
 
   def enqueue_decorated(event, client_ip_address, client_address, client_port, socket)
-    event.set(HOST_FIELD, client_address) unless event.get(HOST_FIELD)
-    event.set(HOST_IP_FIELD, client_ip_address) unless event.get(HOST_IP_FIELD)
-    event.set(PORT_FIELD, client_port) unless event.get(PORT_FIELD)
-    event.set(SSLSUBJECT_FIELD, socket.peer_cert.subject.to_s) if socket && @ssl_enable && @ssl_verify && event.get(SSLSUBJECT_FIELD).nil?
+    event.set(@field_host, client_address) unless event.get(@field_host)
+    event.set(@field_host_ip, client_ip_address) unless event.get(@field_host_ip)
+    event.set(@field_port, client_port) unless event.get(@field_port)
+    event.set(@field_sslsubject, socket.peer_cert.subject.to_s) if socket && @ssl_enable && @ssl_verify && event.get(@field_sslsubject).nil?
     decorate(event)
     @output_queue << event
   end
 
+  # setup the field names, with respect to ECS compatibility.
+  def setup_fields!
+    @field_host       = ecs_select[disabled: "host",                    v1: "[@metadata][tcp][source][name]"        ].freeze
```

Comment:
what about using `source.address` instead 
so the underlying structure could be copied to a top-level and would be ecs-compliant.

Reply:
There are a few reasons that make wholesale copy of this source metadata impractical.

While `source` and `destination` fields do have an `address`, `ip`, and `port`, they are used to describe network _transactions_ captured _in_ an event. Similarly, `client` and `server` have all three fields, but are used to describe the initiator or recipient of the network transaction.

While the name and ip of the event's source _may_ make sense to copy into `host` or `observer`, the `port` is pretty meaningless and doesn't have a home there. When this plugin is run in server mode, the outbound port from the client is typically chosen at random by its host OS, and when the plugin is run in client mode, the port we are connecting to doesn't really describe the _event_.

I chose `name` here because it aligns with both `host` and `observer`.

We _could_ align with the CEF codec, which has a new `device` option that accepts `host` or `observer`, and could populate the relevant bits if this is set?

Reply:
thanks for sharing the details - fine by me to keep as is - prefixed `source.name`